mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -0,0 +1,529 @@
+        let f = self.taker_coin.wait_for_tx_spend(
+            &self.data.taker_payment_hex[..],
+            self.data.taker_payment_lock,
+            self.data.taker_coin_start_block,
+            &None,
+        );
+
+        let tx = match f.compat().await {
+            Ok(t) => t,
+            Err(err) => {
+                return Ok((Some(WatcherCommand::Finish), vec![
+                    WatcherEvent::TakerPaymentWaitForSpendFailed(err.get_plain_text_format().into()),
+                ]));
+            },
+        };
+
+        let tx_hash = tx.tx_hash();
+        info!("Taker payment spend tx {:02x}", tx_hash);
+        let tx_ident = TransactionIdentifier {
+            tx_hex: tx.tx_hex().into(),
+            tx_hash,
+        };
+
+        let secret = match self
+            .taker_coin
+            .extract_secret(&self.data.secret_hash[..], &tx_ident.tx_hex.0)
+        {
+            Ok(bytes) => H256Json::from(bytes.as_slice()),
+            Err(e) => {
+                return Ok((Some(WatcherCommand::Finish), vec![
+                    WatcherEvent::TakerPaymentWaitForSpendFailed(ERRL!("{}", e).into()),
+                ]));
+            },
+        };
+
+        Ok((Some(WatcherCommand::SpendMakerPayment), vec![
+            WatcherEvent::TakerPaymentSpent(TakerPaymentSpentData {
+                transaction: tx_ident,
+                secret,
+            }),
+        ]))
+    }
+
+    async fn spend_maker_payment(&self) -> Result<(Option<WatcherCommand>, Vec<WatcherEvent>), String> {
+        let spend_fut = self.maker_coin.send_taker_spends_maker_payment_preimage(
+            &self.data.taker_spends_maker_payment_preimage,
+            &self.r().secret.0.clone(),
+        );
+
+        let transaction = match spend_fut.compat().await {
+            Ok(t) => t,
+            Err(err) => {
+                if let Some(tx) = err.get_tx() {
+                    broadcast_p2p_tx_msg(&self.ctx, tx_helper_topic(self.maker_coin.ticker()), &tx, &None);
+                };
+
+                return Ok((Some(WatcherCommand::Finish), vec![
+                    WatcherEvent::MakerPaymentSpendFailed(ERRL!("{}", err.get_plain_text_format()).into()),
+                ]));
+            },
+        };
+
+        broadcast_p2p_tx_msg(
+            &self.ctx,
+            tx_helper_topic(self.maker_coin.ticker()),
+            &transaction,
+            &None,
+        );
+
+        let tx_hash = transaction.tx_hash();
+        info!("Maker payment spend tx {:02x}", tx_hash);
+        let tx_ident = TransactionIdentifier {
+            tx_hex: transaction.tx_hex().into(),
+            tx_hash,
+        };
+
+        Ok((Some(WatcherCommand::Finish), vec![WatcherEvent::MakerPaymentSpent(
+            tx_ident,
+        )]))
+    }
+}
+
+impl AtomicSwap for Watcher {
+    fn locked_amount(&self) -> Vec<LockedAmount> { unimplemented!() }
```

Comment:
Also `Watcher::locked_amount` can be invoked on [get_locked_amount](https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1/mm2src/mm2_main/src/lp_swap.rs#L419) that is commonly used to check the available balance for swaps.
So I think that:
1) Watcher swaps shouldn't be stored at `SwapsContext::running_swaps`.
2) `Watcher` shouldn't implement `AtomicSwap` trait as it's not actually an atomic swap.